src/tallies/TallyBase.C

```diff
@@ -48,6 +48,9 @@ TallyBase::validParams()
       "this same trigger is applied to all scores.");
   params.addRangeCheckedParam<std::vector<Real>>(
       "trigger_threshold", "trigger_threshold > 0", "Threshold for the tally trigger");
+  params.addParam<std::vector<bool>>(
+      "trigger_ignore_zeros",
```

Comment:
My gut reaction is to assign a default to everything, so that there's kind of a suggestion to beginner users on what these parameters typically are.

Reply:
Then we also don't need to check `isParamValid`, because it will have a value even if the user doesn't set it

Reply:
Sounds good. I've added a default of `false` to `trigger_ignore_zeros` and added functionality to allow users to specify a single value of `trigger_ignore_zeros` which is than applied to all tally scores.